src/expr/src/scalar/func/impls/string.rs

```diff
@@ -956,6 +958,63 @@ impl fmt::Display for RegexpMatch {
     }
 }
 
+#[derive(Ord, PartialOrd, Clone, Debug, Eq, PartialEq, Serialize, Deserialize, Hash, MzReflect)]
+pub struct RegexpSplitToArray(pub Regex);
+
+impl LazyUnaryFunc for RegexpSplitToArray {
+    fn eval<'a>(
+        &'a self,
+        datums: &[Datum<'a>],
+        temp_storage: &'a RowArena,
+        a: &'a MirScalarExpr,
+    ) -> Result<Datum<'a>, EvalError> {
+        let haystack = a.eval(datums, temp_storage)?;
+        if haystack.is_null() {
+            return Ok(Datum::Null);
+        }
+        regexp_split_to_array_re(haystack.unwrap_str(), &self.0, temp_storage)
+    }
+
+    /// The output ColumnType of this function
+    fn output_type(&self, _input_type: ColumnType) -> ColumnType {
+        ScalarType::Array(Box::new(ScalarType::String)).nullable(true)
```

Comment:
You can set `.nullable` to `input_type.nullable`. (Provided that `introduces_nulls` being `false` is correct. From looking at the implementation, it seems correct to me: We return null only when the input is null.)

Reply:
Thanks, done.